### ISSUE TITLE
OTVortex fails parallel/serial comparison

### ISSUE THREAD

**Issue by AlexJamesWright** [AlexJamesWright/METHOD#24]
The OTVortex is a much more stringent test to assess if the MPI implementation is working properly, the state is not homogenous anywhere and so will test if boundary conditions are correct more so than a BrioWu, for example.

In commit 595030b7f959d49f342ea241ecb2071038e7a083 on branch OTVParallelTests, the comparison between MPI and serial fails, suggesting there is still a bug somewhere in ParallelBoundaryConditions. The comparison for all other tests is exact (TOL of 1e-200 still passes) so I'm sure this isn't an accumulation of numerical errors that is causing OTVortex to fail.

For these tests I have taken the RandomKHInstability tests, and changed only the initial data to OTVortex. Playing around, I noticed that the failed cells were always (I think) within data->Ng cells of a boundary (subdomain or physical).

Any ideas where this might be coming from?

**Comment by aniabrown** [comment]
This is likely because the OTVortex initialisation hasn't been implemented in parallel -- there will be a problem if any global variables are referred to there.

**Comment by aniabrown** [comment]
Hm, no, I don't see anything global there. Will take a closer look.

**Comment by AlexJamesWright** [comment]
So if you check out the latest commit on OTVParallelTests and run the test_rk2 and test_parallel_rk2, I've included the outflow and periodic OTV. Most of the state vectors fail after just one iteration (not zero as I thought previously). Also, I've checked the output and the were not smart guessing failures for these tests

**Comment by aniabrown** [comment]
This was mainly an issue with floating point rounding error for a chaotic system. A different method of initialising x, y, and z in the parallel version was creating slightly different initial conditions, which seemed to be blowing up. I have updated the x, y, z init to be bitwise identical to the serial version. 

In addition, RKSplitSrmhdOutflowOTVSF in the parallel test cases mistakenly wrote files to RK2SrmhdOutflowOTVSF, which was overwriting the RK2 results. I've commented the RKSplitSrmhdOutflowOTVSF test for now as there are no serial tests -- I don't know how much of RKSplit is tested with the RK2 tests so not sure if we should bother creating RKSplit tests for OTV. With RKSplitSrmhdOutflowOTVSF commented out and the change above all tests should now pass on this branch.
